keylime/allowlistdbproxy.py

```diff
@@ -103,6 +110,12 @@ def check_update(self) -> bool:
                         self.allowlist_checksum = self.verifier_allowlist_db.checksum  # type: ignore
 
                         self.verifier_allowlist = AllowlistEntry.__copy_verifier_allowlist(self.verifier_allowlist_db)
+                    else:
+                        # Refresh anyway...
+                        logger.info(f"ima_policy BEFORE: {self.verifier_allowlist_db.ima_policy}")
+                        self.session.refresh(self.verifier_allowlist_db)
+                        logger.info(f"ima_policy AFTER : {self.verifier_allowlist_db.ima_policy}")
```

Comment:
It does seem to be the case that the checksum did not get updated but the ima_policy changed: @maugustosilva 
```
Nov 17 00:24:37 ip-172-31-19-1.us-east-2.compute.internal keylime_verifier[59508]: 2022-11-17 00:24:37.660 - keylime.allowlistdbproxy - INFO - ima_policy BEFORE: {"allowlist": {"meta": {"version": 5, "checksum": "d8d581d93b1421ec6874b84369e880eb1a7c1275ad5f95b13faa6696e8637066", "timestamp": "2022-11-17 00:24:24.121668", "generator": "keylime-legacy-format-upgrade"}, "release": 0, "hashes": {"boot_aggregate": ["0000000000000000000000000000000000000000000000000000000000000000"]}, "keyrings": {}, "ima": {"ignored_keyrings": [], "log_hash_alg": "sha1", "dm_policy": null}, "ima-buf": {}}, "exclude": ["/afs(/.*)?", "/bin(/.*)?", "/boot(/.*)?", "/dev(/.*)?", "/etc(/.*)?", "/home(/.*)?", "/lib(/.*)?", "/lib64(/.*)?", "/media(/.*)?", "/mnt(/.*)?", "/opt(/.*)?", "/proc(/.*)?", "/root(/.*)?", "/run(/.*)?", "/sbin(/.*)?", "/srv(/.*)?", "/sys(/.*)?", "/tmp(/.*)?", "/usr(/.*)?", "/var(/.*)?", "/sysroot/etc/fstab", "/dracut-state.sh", "/keylime-tests/basic-attestation-on-localhost-pbI7Q(/.*)?", "/keylime-tests/basic-attestation-on-localhost-s01Ko(/.*)?", "/keylime-tests/basic-attestation-with-custom-certificates-R1GzJ(/.*)?", "/keylime-tests/basic-attestation-with-ima-signatures-y49Ml(/.*)?", "/keylime-tests/basic-attestation-without-mtls-6ovPZ(/.*)?", "/keylime-tests/basic-attestation-with-unpriviledged-agent-9xbL6(/.*)?", "/keylime-tests/install-rpm-with-ima-signature-S6TAh(/.*)?", "/keylime-tests/keylime-non-default-ports-uLO6I(/.*)?"]}
Nov 17 00:24:37 ip-172-31-19-1.us-east-2.compute.internal keylime_verifier[59508]: 2022-11-17 00:24:37.661 - keylime.allowlistdbproxy - INFO - ima_policy AFTER : {"allowlist": {"meta": {"version": 5, "checksum": "d8d581d93b1421ec6874b84369e880eb1a7c1275ad5f95b13faa6696e8637066", "timestamp": "2022-11-17 00:24:37.592173", "generator": "keylime-legacy-format-upgrade"}, "release": 0, "hashes": {"boot_aggregate": ["0000000000000000000000000000000000000000000000000000000000000000"]}, "keyrings": {}, "ima": {"ignored_keyrings": [], "log_hash_alg": "sha1", "dm_policy": null}, "ima-buf": {}}, "exclude": ["/afs(/.*)?", "/bin(/.*)?", "/boot(/.*)?", "/dev(/.*)?", "/etc(/.*)?", "/home(/.*)?", "/lib(/.*)?", "/lib64(/.*)?", "/media(/.*)?", "/mnt(/.*)?", "/opt(/.*)?", "/proc(/.*)?", "/root(/.*)?", "/run(/.*)?", "/sbin(/.*)?", "/srv(/.*)?", "/sys(/.*)?", "/tmp(/.*)?", "/usr(/.*)?", "/var(/.*)?", "/sysroot/etc/fstab", "/dracut-state.sh", "/keylime-tests/basic-attestation-on-localhost-pbI7Q(/.*)?", "/keylime-tests/basic-attestation-on-localhost-s01Ko(/.*)?", "/keylime-tests/basic-attestation-with-custom-certificates-R1GzJ(/.*)?", "/keylime-tests/basic-attestation-with-ima-signatures-y49Ml(/.*)?", "/keylime-tests/basic-attestation-without-mtls-6ovPZ(/.*)?", "/keylime-tests/basic-attestation-with-unpriviledged-agent-9xbL6(/.*)?", "/keylime-tests/install-rpm-with-ima-signature-S6TAh(/.*)?", "/keylime-tests/keylime-non-default-ports-uLO6I(/.*)?", "/keylime-tests/keylime_tenant-commands-on-localhost-zFCBQ(/.*)?"]}
```

Reply:
The failure occurred in this test run: https://artifacts.dev.testing-farm.io/88e306a5-2d73-4b59-ab93-05df9bdf733d/

Reply:
The checksum is always calculated over the `allowlist` only. If any other aspect of the IMA policy, such as the `excludelist` was changed, a new checksum is (currently) not recalculated.